specification/storagepool/resource-manager/Microsoft.StoragePool/preview/2020-03-15-preview/storagepool.json

```diff
@@ -860,19 +1085,53 @@
         }
       ]
     },
+    "IscsiTargetCreate": {
+      "type": "object",
+      "description": "Payload for iSCSI Target create or update requests.",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties for iSCSI Target create request.",
+          "$ref": "#/definitions/IscsiTargetCreateProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "IscsiTargetUpdate": {
+      "type": "object",
+      "description": "Payload for iSCSI Target update request.",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties for iSCSI Target update request.",
+          "$ref": "#/definitions/IscsiTargetUpdateProperties"
+        }
+      }
+    },
     "IscsiTargetProperties": {
       "type": "object",
-      "description": "Properties of iSCSI Target request payload.",
+      "description": "Response properties for iSCSI Target operations.",
       "required": [
         "tpgs",
-        "targetIqn"
+        "targetIqn",
+        "provisioningState",
```

Comment:
Make sure to add to the required list those properties that the user must set, unless there's a different motivation to add these (ProvisioningState and Status) under required?

Reply:
Yeah -- same as above. We use IscsiTarget and the nested properties IscsiTargetProperties as the response payload returned by the RP. So the customers shouldn't be using it to make create/update requests.

They use IscsiTargetCreate/IscsiTargetUpdate for PUT/PATCH requests on the resource